core-ui/src/main/java/com/maxshop/adapter/DataBindingRecyclerAdapter.kt

```diff
@@ -0,0 +1,42 @@
+package com.maxshop.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.recyclerview.widget.ListAdapter
+import com.maxshop.adapter.callback.DiffCallback
+import com.maxshop.adapter.viewHolder.BindingViewHolder
+import com.maxshop.model.RecyclerItem
+
+class DataBindingRecyclerAdapter : ListAdapter<RecyclerItem, BindingViewHolder>(DiffCallback()) {
+    override fun onCreateViewHolder(
+        parent: ViewGroup,
+        viewType: Int
+    ): BindingViewHolder {
+        val inflater = LayoutInflater.from(parent.context)
+        val binding: ViewDataBinding =
+            DataBindingUtil.inflate(inflater, viewType, parent, false)
```

Comment:
Please add this line after you create the binding:
`binding.lifecycleOwner = parent.findViewTreeLifecycleOwner()`

Reply:
fixed